oslc4j-jena-provider/src/main/java/org/eclipse/lyo/oslc4j/provider/jena/ordfm/RDFTypes.java

```diff
@@ -0,0 +1,179 @@
+package org.eclipse.lyo.oslc4j.provider.jena.ordfm;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ClassInfoList;
+import io.github.classgraph.ScanResult;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.vocabulary.RDF;
+import org.eclipse.lyo.oslc4j.core.annotation.OslcResourceShape;
+import org.eclipse.lyo.oslc4j.core.model.TypeFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * @author rherrera
+ */
+public class RDFTypes {
+
+    /**
+     * The logger of this class.
+     */
+	private static final Logger LOGGER = LoggerFactory.getLogger(RDFTypes.class);
+
+    /**
+     * The set of scanned packages.
+     */
+    static final Set<String> SCANNED_PACKAGES = new HashSet<>();
+
+    /**
+     * The RDFs-Classes types mapping.
+     */
+    static final Map<String, List<Class<?>>> TYPES_MAPPINGS = new HashMap<>();
+
+    /**
+     * Scans a package (recursively) searching for classes annotated with
+     * {@link OslcResourceShape} and mapping them with an {@link RDF#type} built
+     * by the {@link TypeFactory#getQualifiedName(java.lang.Class)} method.
+     * @param pkg the package to scan.
+     */
+    public synchronized static void mapPackage(Package pkg) {
+        String packageName = pkg.getName();
+        if (SCANNED_PACKAGES.contains(packageName)) {
+            LOGGER.debug("> package {} already scanned", packageName);
+        } else {
+            int counter = 0;
+            LOGGER.debug("> scanning package {}", packageName);
```

Comment:
TRACE?

Reply:
I do normally use DEBUG for the first and last logging messsages of a public method, then I use TRACE for the first and last logging messages of private methods (when useful) or for internal processing of public methods. In this case, I know registering a package may be considered as an internal operation of the overall process, but still this method needs to be public and I'd like to keep these as DEBUG if you don't mind.

Reply:
Interesting approach. Let's try it.